### PR TITLE
Deprecate tpm1 and deep quote

### PR DESCRIPTION
**Pull request by lukehinds**
This change removes all deep quote code and tpm1 functionality

Various tpm values such as `tpm_version` are also removed.

Signed-off-by: Luke Hinds <lhinds@redhat.com>

### CODE REVIEW

File: keylime/tenant.py [hunk 1]
```diff
@@ -434,19 +436,18 @@ def check_ek(self, ek, ekcert, tpm):
                 logger.warning(
                     "No EK cert provided, require_ek_cert option in config set to True")
                 return False
-            elif not tpm.verify_ek(base64.b64decode(ekcert), ek):
+            elif not self.tpm_instance.verify_ek(base64.b64decode(ekcert), ek):
```
Comment (kaifeng) [inline keylime/tenant.py:439]: I suggest we move verify_ek (as well as other hardware unrelated implementations) out of tpm, tenant doesn't need a hardware tpm to run.

**Review by kaifeng — COMMENTED** [review]
We also need to remove tpm_version from the database and agent processing I guess.

File: keylime/tenant.py (hunk 1 above)
Comment (lukehinds) [reply keylime/tenant.py:439]: thanks for noticing that, would explain why my quotes were failing.

on the topic of moving verify_ek we should discuss that in an issue.

**Comment by lukehinds** [comment]
> We also need to remove tpm_version from the database and agent processing I guess.

Yep, I have a migration to do that.

File: keylime/tpm/tpm_main.py [hunk 2]
```diff
@@ -870,6 +869,8 @@ def verify_ek(self, ekcert, ekpem):
             ekcertpem = ek509.get_pubkey().get_rsa().as_pem(cipher=None).decode('utf-8')
 
             # Make sure given ekcert is for their ek
+            print('ekpem: ', str(ekpem))
```
Comment (mpeters) [inline keylime/tpm/tpm_main.py:872]: Was this left in for debugging? Should it be sent to the logger instead?

File: test/test_restful.py [hunk 3]
```diff
@@ -450,35 +442,36 @@ def test_011_reg_agent_activate_put(self):
...
-            cert="",
-            verify=False
-        )
-
-        self.assertEqual(response.status_code, 200, "Non-successful Registrar agent vActivate return code!")
-        json_response = response.json()
-
-        # Ensure response is well-formed
-        self.assertIn("results", json_response, "Malformed response body!")
+    # @unittest.skipIf(not vtpm, "Registrar's PUT /v2/agents/{UUID}/vactivate only for vTPMs!")
```
Comment (mpeters) [inline test/test_restful.py:445]: Why is this being commented out? Should it be deleted instead?

**Review by mpeters — CHANGES_REQUESTED** [review]
This is missing the DB migration change for deleting the tpm_version from the DB

File: test/test_restful.py (hunk 3 above)
Comment (lukehinds) [reply test/test_restful.py:445]: It should , was just testing first...will remove this for now

File: keylime/tpm/tpm_main.py (hunk 2 above)
Comment (lukehinds) [reply keylime/tpm/tpm_main.py:872]: ack, this is debug residue.

**Review by mpeters — APPROVED** [review]
lgtm

**Comment by nabilschear** [comment]
sadness for all that code.  But, life must move on!

**Pull request merged**